README.md

```diff
@@ -277,10 +277,6 @@ See [trogdor/README.md](trogdor/README.md).
 
 See [vagrant/README.md](vagrant/README.md).
 
-### Release Kafka ###
-
-See [release/README.md](release/README.md).
-
 ### Official Documentation ###
```

Comment:
@johnny94 , Chia-Ping mentioned Official Documentation could be removed too. Do you want to include it in this PR?

https://github.com/apache/kafka/pull/19420#discussion_r2045913751

Reply:
I missed it. Thanks.